docs/4.3. Behaviour - Tokens.md

```diff
@@ -96,39 +96,66 @@ _Contains information particular to the NMOS API the token is intended for_
 
 ```json
 "x-nmos-api": {
-  "register": [],
-  "connection": []
+  "register": {},
+  "connection": {}
 }
 ```
 
-The `value` for each NMOS API indicates the access rights of the given user for that specific API. The value required
-by the Resource Server MUST exactly match one of the values in the array. General access rights include:
+##### The Access Permissions Object
+
+The value of each NMOS API key is the access permissions object for the given user for that specific API. The access
+permission object consists of `key:value` pairs in which the `key` denotes a specific access right. General access
+rights include:
 *   **write** - ability to insert a new, or edit an existing, resource. This takes the form of *POST*, *PUT*,
     *PATCH* and *DELETE* HTTP request verbs.
 *   **read** - ability to retrieve the resource. This takes the form of *GET*, *OPTIONS* and *HEAD* HTTP request
     verbs.
 
-If a value of "write" is given in an access array, and the intention is that the given user also has "read" access to
-a given API, the value of "read" MUST also occur in the array.
+If an access right value of "write" is given in the access permission object, and the intention is that the given user
+also has "read" access to the given API, a value of "read" MUST also be present in the access permissions object.
+
+Individual AMWA specifications MAY specify additional access rights specific to the individual API. These MUST be
+registered in the [AMWA NMOS Parameter Registers].
 
-Individual AMWA specifications MAY specify additional values specific to the individual API. These MUST be registered
-in the [AMWA NMOS Parameter Registers].
+The `value` corresponding to each access right is a JSON array containing path specifiers that the user is permitted
+to perform the specific request against. Wildcards are permitted in the path specifiers. The path specifiers MUST use
+dot notation to signify specific sub-paths of the endpoint. For NMOS API's following the standard pattern, all path
+specifiers MUST start _after_ the API version.
+
+For example, authorizing a user to read all _single resources_ and make connections between all _single senders_ at a
+Connection Management interface would result in an access permissions object of:
+
+```JSON
+"connection": {
+  "read": ["single.*"],
+  "write": ["single.senders.*"]
+}
+```
 
 An example of a `x-nmos-api` claim is shown below. This would permit querying an IS-04 Registry, and making
-connections via an IS-05 connection management interface, but would not permit the user to register a resource with
-an IS-04 Registry.
+_single_ connections via an IS-05 connection management interface, but would not permit the user to register a resource
+with an IS-04 Registry.
 
 ```json
 "x-nmos-api": {
-  "register": ["read"],
-  "query": ["read", "write"],
-  "connection": ["write", "read"]
+  "register": {
+    "read": ["*"],
+    "write": []
```

Comment:
Are 'read' and 'write' mandatory with optional array contents? We should state this explicitly if so. I do wonder however if we should allow read/write to be completely omitted in case other claims are defined in the future, otherwise the token will be a little bloated.